### ISSUE TITLE
Enhancement/Question: Is stricter typing out of scope for this project?

### ISSUE THREAD

**Issue by justuswilhelm** [anephenix/sarus#320]
First of all, I would like to say my thanks for creating a library that works so well out of the box. I have tried creating a WS connection manager myself, until hitting too many edge cases.

The TypeScript dom API sees WebSocket messages as optionally typable (see https://github.com/microsoft/TypeScript/blob/d3513ee5c4960467c458c5d567437194aa3e828c/src/lib/dom.generated.d.ts#L956)

In my case, I receive WebSocket messages that are simple strings, and I am initializing the eventListeners like so:

```typescript
    const sarus = new Sarus({
        url,
        eventListeners: {
// other handlers
            message: [
                (theMessage: MessageEvent<string>) => {
                    console.log(theMessage);
                },
            ],
        },
    });
```

If I leave the type variable inside MessageEvent empty, `data` will be any for the TypeScript compiler. It would be great if I could create a Sarus instance like so:

```typescript
const sarus = new Sarus<string>(...)
```

Upon which I am forced to create message handlers that have to handle `MessageEvent<string>` and not accidentally for `number` and so on instead.

What do you think? Is this out of scope?

One side note, the `EventListenersInterface` might be easier to use if all the attributes like `open`, `message`, and so on are optional with `?`. Then, when creating a new Sarus instance, I don't have to create no-op handlers for the ones that I am not interested in.

I am pasting the MessageEvent interface here for reference, taken from `lib.dom.d.ts`:

```typescript
/** A message received by a target object. */
interface MessageEvent<T = any> extends Event {
    /** Returns the data of the message. */
    readonly data: T;
    /** Returns the last event ID string, for server-sent events. */
    readonly lastEventId: string;
    /** Returns the origin of the message, for server-sent events and cross-document messaging. */
    readonly origin: string;
    /** Returns the MessagePort array sent with the message, for cross-document messaging and channel messaging. */
    readonly ports: ReadonlyArray<MessagePort>;
    /** Returns the WindowProxy of the source window, for cross-document messaging, and the MessagePort being attached, in the connect event fired at SharedWorkerGlobalScope objects. */
    readonly source: MessageEventSource | null;
    /** @deprecated */
    initMessageEvent(type: string, bubbles?: boolean, cancelable?: boolean, data?: any, origin?: string, lastEventId?: string, source?: MessageEventSource | null, ports?: MessagePort[]): void;
}
```

**Comment by paulbjensen** [comment]
Hi @justuswilhelm, apologies for the delay I've just noticed the message now. I will have a look at it later today and get back to you.

**Comment by paulbjensen** [comment]
I will have a go at making the eventListener events optional as a starter and then evaluate the prospect of the 1st proposed change.

**Comment by paulbjensen** [comment]
Just published v0.4.5 to support passing in partial event listeners to the Sarus class upon instantiation, that way you can pass in an event handler for message, but not open, error and close for example.

**Comment by paulbjensen** [comment]
Hi @justuswilhelm, I just published v0.4.6 which has the change removing the string typing on messages sent via WebSockets. This should now bring it in line with the TypeScript typing (the only caveat is that I used the `unknown` type versus the `any` type in this case).

Let me know if that's working good for you.

**Comment by paulbjensen** [comment]
Will mark this as closed.
